### PR TITLE
Feature/improve llm pre processing mechanism

### LINKED ISSUE #88: Improve pre-processing (add context, reformat) mechanism of UserInput
### Acceptance criteria
- Pre-processing mechanisms are improved.
- User input is enriched with relevant context.
- Input is reformatted to ensure compatibility with LLM.
- Mechanism is tested for accuracy and performance.

### LINKED ISSUE #89: Improve LLM interaction mechanism to process UserInput and generate a LinktaFLow
### Acceptance criteria
- Implement a mechanism to ensure the generated LinktaFlow adheres to the desired format.
- Generated LinktaFlow is validated against the desired format.
- Validation errors are handled gracefully.
- LinktaFlow is formatted correctly before storage.

### PR DESCRIPTION
**Pull request by zhiyi15**
## Description
- updated LLM processing mechanism from text-only to multi turn(chat)
- updated GoogleGenAI model to `gemini-1.5-flash` and `google/generative-ai` package
- implemented generativ_config, safetySettings, systemInstructions and maxOutputToken for GenAI model improved overall developer control
- added `generateInitialResponse` controller in userInput route  
##### ***Note: I have created another PR to refactor `userInputController` #170, but this PR 167 needs to be merged first

Fixes # (issue)[#88 ](https://github.com/orgs/Linkta-org/projects/6/views/1?pane=issue&itemId=63064032)& [#89](https://github.com/orgs/Linkta-org/projects/6/views/1?pane=issue&itemId=63064032)

## Type of change

- [ ]  🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ✨  New feature (non-breaking change which adds functionality)
- [x] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 Documentation update
- [ ] Other 🗒️ ______________

## How Has This Been Tested?

when the errors are resolved in userInputController, LLM workflow works as desired. 
image

## Checklist:

- [x] I have performed a self-review of my own code.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have reviewed the [**Security Best Practices Document**](./docs/SECURITY_BEST_PRACTICES.md).
- [ ] I have followed the [**Naming Conventions Guide**](./docs/NAMING_CONVENTIONS_GUIDE.md) for my changes.
- [ ] I have ensured that my pull request title is descriptive.

### CODE REVIEW

**Review by colinvandergraaf — APPROVED** [review]
Looks Good!  Would love to hear recap of how this improves things next stand up

**Review by ChTiSh — APPROVED** [review]
Thank you @zhiyi15, looks good.

File: server/controllers/genAiController.ts [hunk 1]
```diff
@@ -1,149 +1,70 @@
...
-        err
-      );
-
-      return next(methodError);
     }
+  } catch (err: unknown) {
+    const methodError = createError(
+      'generateInitialResponse',
+      'genAIController',
+      'Erro generating response from AI.',
```
[the remark this answers is not in the document]
Comment (zhiyi15) [reply server/controllers/genAiController.ts:29]: hahah good eye!!!

File: server/models/GeminiModel.ts [hunk 2]
```diff
@@ -1,110 +1,84 @@
...
+    '#CONTEXT#\n\nI am a visual learner. I need you to process and brainstorm input subject for me\n#############\n\n# OBJECTIVE#\n\nI want you use to analysis input subject and create a tree-like data structure using parent and child nodes to represents the relationships. Expand at least 4-level in depth. Use this step-by-step process :\n\n1. define the topic based on input subject as top tree node\n2 .create a tree-like data structure to represent the process or brainstorm information, where each node represents a subject. Assign a unique ‘id’ to each node based on its level in the tree, and include ‘data’ property containing a \'label\' property with the maximum of three keywords that describe the subject as its value.\n3.convert the tree like data structure into a JSON object containing two properties ‘nodes’ and ‘edges’ in the same manner as React Flow’s Node and Edge type.\n#############\n\n# STYLE#\n\nJSON\n#############\n\n# TONE#\n\nProfessional, technical\n#############\n\n# AUDIENCE#\n\nAdult learner; ADHD learner\n#############\n\n# RESPONSE: a JSON object#\n\n\'{"nodes":[{id, data:{label}}],"edges":[{id, source, target}]}\'\n#############\n\n# START ANALYSIS#',
```
[the remark this answers is not in the document]
Comment (zhiyi15) [reply server/models/GeminiModel.ts:57]: lets goooooooo!!

**Pull request merged**